src/Sulu/Bundle/ContentBundle/Resources/public/js/main.js

```diff
@@ -68,22 +68,50 @@ define([
                         locale: data.locale
                     };
                 },
-                function (key) {
+                function(key) {
                     if (key.indexOf('page_') === 0) {
                         return 'page';
                     }
                 }
             );
 
-            function getContentLanguage() {
-                return sandbox.sulu.getUserSetting('contentLanguage') || Object.keys(Config.get('sulu-content').locales)[0];
+            /**
+             * Returns the the language of the content which is displayed at the beginning.
+             * It first look for the last chosen language for this webspace. If no such
+             * language exists, the function looks for the last chosen language over all
+             * webspaces and if that language is possible in with the given webspace. If
+             * not language can be found, the default language is returned.
+             *
+             * @param {String} webspace The webspace key
+             * @returns {string} The locale of the content
+             */
+            function getContentLanguage(webspace) {
+                var webspaceLanguage = sandbox.sulu.getUserSetting(webspace + '.contentLanguage');
+                if (!!webspaceLanguage) {
+                    return webspaceLanguage;
+                }
+
+                var lastLanguage = sandbox.sulu.getUserSetting('contentLanguage');
+                if (!!lastLanguage && !!Config.get('sulu-content').locales[webspace][lastLanguage]) {
+                    return lastLanguage;
+                }
+
+                if (!Config.get('sulu-content').locales[webspace]
+                    || Config.get('sulu-content').locales[webspace].length === 0) {
+                    app.logger.error('Webspace "' + webspace + '" has no defined locale');
+                }
+
+                // the default locale of a webspace is always on the first position
+                var defaultLocale = Object.keys(Config.get('sulu-content').locales[webspace])[0];
```

Comment:
I think in terms of efficiency it would be the same or even a bit better if you use only one variables, and override its value in this function instead of allocating memory three times.